lib/nginx.js

```diff
@@ -0,0 +1,130 @@
+var nginxConf = require('./nginx-conf').genNginxConf;
+var configHash = require('./nginx-conf').getConfigHash;
+var path = require('path');
+var debug = require('debug')('strong-nginx-controller:nginx');
+var childProcess = require('child_process');
+var async = require('async');
+
+function Nginx(options) {
+  this._app = options.app;
+  this._nginxPath = options.nginxPath;
+  this._baseDir = options.baseDir;
+  this._nginxConf = nginxConf.bind(null, options);
+  this._configHash = configHash.bind(null, options);
+}
+
+/**
+ * Generate the nginx config and start nginx.
+ */
+function start(callback) {
+  async.series([
+    this._nginxConf,
+    this._cmd.bind(this, 'start')
+  ], callback);
+}
+Nginx.prototype.start = start;
+
+var MAX_DELAY = 1000 * 1000; // 1sec
```

Comment:
Is this intentionally in microseconds? It appears to be used below as milliseconds (what `Date.now()` returns), which would just be `1 * 1000`, wouldn't it?